File: Editorial/135_Candy/C/3_Using_one_array.c
int candy(int* ratings, int ratingsSize) {
    int* candies = (int*)malloc(ratingsSize * sizeof(int));
    for (int i = 0; i < ratingsSize; i++) {
        candies[i] = 1;
    }
    for (int i = 1; i < ratingsSize; i++) {
        if (ratings[i] > ratings[i - 1]) {
            candies[i] = candies[i - 1] + 1;
        }
    }
    int sum = candies[ratingsSize - 1];
    for (int i = ratingsSize - 2; i >= 0; i--) {
        if (ratings[i] > ratings[i + 1] && candies[i] <= candies[i + 1]) {
            candies[i] = candies[i + 1] + 1;
        }
        sum += candies[i];
    }
    free(candies);
    return sum;
}

File: Editorial/0101-0200/0135_Candy/Go/3_Using_one_array.go
func candy(ratings []int) int {
    candies := make([]int, len(ratings))
    for i := range candies {
        candies[i] = 1
    }
    for i := 1; i < len(ratings); i++ {
        if ratings[i] > ratings[i-1] {
            candies[i] = candies[i-1] + 1
        }
    }
    sum := candies[len(candies)-1]
    for i := len(ratings) - 2; i >= 0; i-- {
        if ratings[i] > ratings[i+1] {
            candies[i] = max(candies[i], candies[i+1]+1)
        }
        sum += candies[i]
    }
    return sum
}

func max(x, y int) int {
    if x > y {
        return x
    }
    return y
}
